src/cli/commands/start.ts

```diff
@@ -6,19 +6,18 @@ import shell from "shelljs";
 import builder from "../../builder";
 import { Dashboard } from "../../dashboard";
 import { createRuntimeHost } from "../../runtimeHost";
-import { GithubActionSWAConfig, isPortAvailable, parseUrl, readConfigFile } from "../../utils";
+import { GithubActionSWAConfig, isAcceptingTcpConnections, isHttpUrl, isPortAvailable, parseUrl, readConfigFile } from "../../utils";
 import { DEFAULT_CONFIG } from "../config";
 
 export async function start(startContext: string, program: CommanderStatic) {
-  if (startContext.startsWith("http")) {
+  if (isHttpUrl(startContext)) {
     // start the emulator and proxy app to the provided uri
-    let { host, port } = parseUrl(startContext);
-    port = port || 4280;
+    let { hostname, port } = parseUrl(startContext);
 
     // make sure host and port are available
     try {
-      const portAvailable = await isPortAvailable({ port, host });
-      if (portAvailable) {
+      const appListening = await isAcceptingTcpConnections({ port, host: hostname });
```

Comment:
`isPortAvailable()` was not doing the right thing on my machine. It returned `true` even though my create-react-app dev server was already running. Not sure why. I had to make this change to continue testing.